common.go

```diff
@@ -37,3 +39,19 @@ func cStringArray(stringList []string) ([]*C.char, func()) {
 		}
 	}
 }
+
+// cError creates an error value from a TileDB error.
+func cError(err *C.tiledb_error_t) error {
+	var str *C.char
+	var msg string
+
+	switch C.tiledb_error_message(err, &str) {
+	case C.TILEDB_OK:
+		msg = C.GoString(str)
+	case C.TILEDB_OOM:
+		msg = "out of memory error while retrieving message"
```

Comment:
Nit: I'd be very specific here because otherwise it may be difficult to tell apart a core error message from an error while retrieving core's error message.
```suggestion
		msg = "out of memory error while retrieving TileDB error message"
```